internal/metrics/metadata.go

```diff
@@ -0,0 +1,102 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package metrics
+
+import (
+	"errors"
+	"sync"
+
+	"go.opentelemetry.io/otel"
+	api "go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/sdk/metric"
+
+	"github.com/envoyproxy/gateway/api/v1alpha1"
+	log "github.com/envoyproxy/gateway/internal/logging"
+)
+
+var (
+	meter = func() api.Meter {
+		return otel.GetMeterProvider().Meter("envoy-gateway")
+	}
+
+	metricsLogger = log.DefaultLogger(v1alpha1.LogLevelInfo).WithName("metrics")
+)
+
+func init() {
+	otel.SetLogger(metricsLogger.Logger)
+}
+
+// MetricType is the type of a metric.
+type MetricType string
+
+// Metric type supports:
+// * Counter: A Counter is a simple metric that only goes up (increments).
+//
+// * Gauge: A Gauge is a metric that represent
+// a single numerical value that can arbitrarily go up and down.
+//
+// * Histogram: A Histogram samples observations and counts them in configurable buckets.
+// It also provides a sum of all observed values.
+// It's used to visualize the statistical distribution of these observations.
```

Comment:
combine with L32?

Reply:
Is this a blocker? Can we do it with a follow-up?